### ISSUE TITLE
Imported Media location changes on re-import

### ISSUE THREAD

**Issue by cybercop23** [xLightsSequencer/xLights#3754]
NOTE: IF YOU DO NOT ATTACH A SEQUENCE THAT DEMONSTRATES YOUR PROBLEM THEN THERE IS A HIGH PROBABILITY YOUR ISSUE WILL JUST BE CLOSED AND NOT INVESTIGATED.

**Describe the bug**
The location of Imported Media changes after you save the sequence.

**To Reproduce**
(and/or link to a short video showing the problem with audio track describing what you are doing)
1. New Sequence
2. Import From Zip
3. Load Mapping File (not necessary) - Saved folder is the name of the zip file in Imported Media folder
4. Save sequence => *** issue *** I save it under a sub-folder called Sequences
5. Re-Import Effects from Zip
6. At this point in time the import media has changed to the Sequences folder where the sequence was saved and doesn't override files so you end up with the wrong version of videos/images if they happen to have the same name
7. Load Mapping
8. Erase existing effects
9. Check location of Videos/Images => now moved to the new location

**Expected behavior**
A reimport from a zip file should preserve the show folder import media location so any new artifacts would be placed in the proper folder and not cause collisions when the files have the same names.

**Screenshots**
Video attached

**Versions (please complete the following information):**
 - OS: Mac, Ventura
 - xLights version 2023.04

**Additional context**
Seems to be related to the fact that I save my sequences in a subfolder (Sequences) under the show folder. Subsequence imports uses that as the starting folder.

**Attachments**
- If associated with a crash please attach/link to a dropbox containing the crash log that was produced
- If it crashed but no crash log was produced please use Tools Menu/Package Logs and attach/link to a dropbox containing the resulting zip file
- If the error relates to rendering an effect or a crash while sequencing please use Tools Menu/Package Sequence and attach/link to a dropbox containing the resulting zip file

https://youtu.be/AT0FzAIWF0k

If you would prefer not to post your sequences publically please say so and be ready to PM them to one of the developers when asked.

**Comment by dartunder** [comment]
This behavior of not using the ImportedMedia folder if the xsq file is not in the base show folder can cause issues if vendors use the same file names for video, for example.  

```
    if (targetDir.ToStdString() != showFolder) {
        // target xsq is not at the root of show folder, assume user manages show folder
        // with a subfolder per sequence as Gil noted
        mediaBaseFolder = targetDir;
    } else {
        // default images/videos/glediators into new ImportedMedia/<NameOfSrcXsq>/<MediaType>
        mediaBaseFolder = wxString::Format("%s%c%s%c%s", showFolder, PATH_SEP, IMPORTED_MEDIA, PATH_SEP, _xsqName);
    }

    // set the defaults for media sub folders
    _importOptions.SetDir(MediaTargetDir::GLEDIATORS_DIR, wxString::Format("%s%c%s", mediaBaseFolder, PATH_SEP, SUBFLD_GLEDIATORS), true);
    _importOptions.SetDir(MediaTargetDir::IMAGES_DIR, wxString::Format("%s%c%s", mediaBaseFolder, PATH_SEP, SUBFLD_IMAGES), true);
    _importOptions.SetDir(MediaTargetDir::VIDEOS_DIR, wxString::Format("%s%c%s", mediaBaseFolder, PATH_SEP, SUBFLD_VIDEOS), true);
```

**Comment by AzGilrock** [comment]
I just get a headache trying to read and figure out what people are doing.  I use a subdirectory for every sequence and never have problems so if people don’t want to do it that way then I got no sympathy when it doesn’t work.

**Comment by cybercop23** [comment]
> I just get a headache trying to read and figure out what people are doing. I use a subdirectory for every sequence and never have problems so if people don’t want to do it that way then I got no sympathy when it doesn’t work.

RIGHT.. that's the problem. If you import from a zip file (as you should) the vids/images go into the "sequence folder". If you have a ..{Sequnces} folder and that's where you create them, vs one per sequnce (seems overkill), then the vids go into the same Video/Images folder and get overriden when sequencers reuse the name.
The Imported Media folder was meant to "fix" that, but only does it if you have the sequence in the show folder, and not any sub-folders (one or many).

**Comment by cybercop23** [comment]
Closed as this will be handled by #4281 ....
